metrics/core/src/main/java/tech/pegasys/pantheon/metrics/prometheus/PrometheusMetricsSystem.java

```diff
@@ -55,6 +57,8 @@
       new ConcurrentHashMap<>();
   private final Map<String, LabelledMetric<tech.pegasys.pantheon.metrics.OperationTimer>>
       cachedTimers = new ConcurrentHashMap<>();
+  private final Map<String, LabelledMetric<Consumer<Supplier<Double>>>> cachedGauges =
```

Comment:
The issue I am addressing is that the pipeline tasks have gauges inside the task we want to track, but they come and go.  I could put the gauge in a static global ;).  But last gauge wins seems to be a better approach.  I also need to add a remove option for when the task is done so we don't leave gauges hanging.

Reply:
ah, I'd deliberately avoided using gauges anywhere that they'd need to be recreated for that reason.

The other approach would be to use a gauge API like Prometheus uses natively - it's basically a counter with `inc()` and `dec()` calls available.  

I'm generally uncomfortable with the current state of gauges because of this potential for memory leaks so last gauge wins may be an improvement but still not ideal.